components/global/core/io.cellery.observability.api/src/main/java/io/cellery/observability/api/UserAuthenticationAPI.java

```diff
@@ -63,14 +68,19 @@ public Response getTokens(@PathParam("authCode") String authCode) throws APIInvo
                     .setCode(authCode).buildBodyMessage();
 
             OAuthClient oAuthClient = new OAuthClient(new URLConnectionClient());
-            OAuthAccessTokenResponse oAuthResponse = oAuthClient.accessToken(request);
+            OAuthAccessTokenResponse oAuthResponse = oAuthClient.accessToken(oAuthClientRequest);
             JSONObject jsonObj = new JSONObject(oAuthResponse.getBody());
             Map<Object, Object> responseMap = new HashMap<>();
 
-            responseMap.put(Constants.ACCESS_TOKEN, oAuthResponse.getAccessToken());
+            String[] tokenParts = splitToken(oAuthResponse.getAccessToken());
+            responseMap.put(Constants.ACCESS_TOKEN, tokenParts[0]);
             responseMap.put(Constants.ID_TOKEN, jsonObj.get(Constants.ID_TOKEN));
 
-            return Response.ok().entity(responseMap).build();
+
+            String cookieDomain = request.getHeader(HttpHeaders.HOST).split(":")[0];
+            NewCookie cookie = new NewCookie("oAuth-Session", tokenParts[1],
```

Comment:
Can we use different name for the cookie.. ? Usually we don't use names which implies the usage when cookies are written. Also you may define this as a constant.

Reply:
Yes. I will change them accordingly.